### PR TITLE
Carbon's Power Laser (WIP)

### PR DESCRIPTION
**Pull request by KayeArray**
Unfinished, we need his sprites and needs a full review.

### CODE REVIEW

File: hippiestation/code/modules/power/power_laser.dm [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+	stat = POWEROFF // starts offline
+	layer = FLY_LAYER
+	var/amount_emitting = 0
+	var/max_amount_emitted = 50000
+	var/total_energy_sent = 0
+	var/turf/turf_to_sell_power_to
+	var/datum/beam/pl_beam/beam
+
+/obj/machinery/power/power_laser/Initialize()
+	..()
```
Comment (ghost) [inline hippiestation/code/modules/power/power_laser.dm:27]: . = ..()

**Review by ghost — CHANGES_REQUESTED** [review]

File: hippiestation/code/modules/power/power_laser.dm [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+			STOP_PROCESSING(SSfastprocess, src)
+			QDEL_NULL(beam)
+		else
+			emit_beam()
+
+/obj/machinery/power/power_laser/setDir(newDir)
+	..()
+	turf_to_sell_power_to = get_edge_target_turf(src, dir)
+
+#define LOW_ENERGY 10000
```
Comment (ghost) [inline hippiestation/code/modules/power/power_laser.dm:172]: Put the beam in it's own file

**Comment by Cyberboss** [comment]
PR this upstream!!

File: hippiestation/code/modules/power/power_laser.dm [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+	icon_state = "ptl" // Especially Gannets!
+	desc = "A really big device capable of converting energy in photons and compressing them enough to make a laser beam out of them, capable of travelling enormous distances through space.\
+			This one is used to send this energy to Centcom, to be converted into money that goes inside the station's budget."
+	anchored = TRUE
+	density = TRUE
+	bound_width = 96
+	bound_height = 96
+	speed_process = TRUE
+	use_power = NO_POWER_USE // We deal with it ourself
+	stat = POWEROFF // starts offline
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/power/power_laser.dm:18]: this needs to be defined.... as to where carbon defined it last time I dunno.

File: hippiestation/code/modules/power/power_laser.dm [hunk 4]
```diff
@@ -0,0 +1,233 @@
+//Power laser, a laser that converts energy to money by sending said energy to centcom through a fat laser beam.
+//Protip, don't get in said laserbeam, unless you're a masochist.
+
+#define ENERGY2MONEY 0.01 // 1 credit each 100 W transferred
+#define ENERGY2DAMAGE 10000// energy/thisdefine=damage
+/obj/machinery/power/power_laser
+	name = "power laser"
+	icon = 'hippiestation/goon/icons/pt_laser.dmi' // huge thanks to goon for the sprite!
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/power/power_laser.dm:8]: This does not exist.

File: hippiestation/code/modules/power/power_laser.dm [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+/obj/machinery/power/power_laser/setDir(newDir)
+	..()
+	turf_to_sell_power_to = get_edge_target_turf(src, dir)
+
+#define LOW_ENERGY 10000
+#define MEDIUM_ENERGY 20000
+#define LARGE_ENERGY 30000
+#define ENORMOUS_ENERGY 40000
+/datum/beam/pl_beam
+	icon = 'hippiestation/goon/icons/ptlbeam.dmi'
```
Comment (GuyonBroadway) [inline hippiestation/code/modules/power/power_laser.dm:177]: this also does not exist.

**Review by GuyonBroadway — CHANGES_REQUESTED** [review]
this is what jenkins is crying about.

File: hippiestation/code/modules/power/power_laser.dm (hunk 4 above)
Comment (ghost) [reply hippiestation/code/modules/power/power_laser.dm:8]: You forgot to read

```Unfinished, we need his sprites and needs a full review.```

File: hippiestation/code/modules/power/power_laser.dm (hunk 3 above)
Comment (Carbonhell) [reply hippiestation/code/modules/power/power_laser.dm:18]: it's a tg define if i remember correctly, machinery to be exact

**Comment by ghost** [comment]
Stale PR, reopen whenever you feel like this is ready for review

**Pull request closed without merging**